src/_pytest/python.py

```diff
@@ -911,6 +911,124 @@ def hasnew(obj: object) -> bool:
     return False
 
 
+@final
+@attr.s(frozen=True, auto_attribs=True)
+class IdMaker:
+    argnames: Iterable[str] = attr.Factory(list)
+    parametersets: Iterable[ParameterSet] = attr.Factory(list)
+    idfn: Optional[Callable[[Any], Optional[object]]] = None
+    ids: Optional[List[Union[None, str]]] = None
+    config: Optional[Config] = None
+    nodeid: Optional[str] = None
+
+    def make_unique_test_ids(self) -> List[str]:
```

Comment:
If you can add some docstring for this function that would be nice (don't have to).